pkg/csoclients/csoclients.go

```diff
@@ -231,9 +241,18 @@ func NewHypershiftGuestClients(
 	}
 	c.MonitoringInformer = prominformer.NewSharedInformerFactory(c.MonitoringClient, resync)
 
-	c.OperatorClient = &operatorclient.OperatorClient{
-		Informers: c.OperatorInformers,
-		Client:    c.OperatorClientSet,
+	gvr := operatorv1.SchemeGroupVersion.WithResource("storages")
+	gvk := operatorv1.SchemeGroupVersion.WithKind("Storages")
```

Comment:
gvr should be plural (`storages`) but gvk should be singular (`Storage`).
You have it right in NewClients() above, this change could explain why the hypershift e2e job is failing:
```
    eventually.go:220: observed *v1beta1.HostedCluster e2e-clusters-nb7vg/example-76h7n invalid at RV 108746 after 2s: incorrect condition: wanted ClusterVersionProgressing=False, got ClusterVersionProgressing=True: ClusterOperatorDegraded(Working towards 4.18.0-0.ci.test-2024-11-14-151003-ci-op-sic0h13g-latest: 493 of 612 done (80% complete), waiting up to 40 minutes on storage)
```

Reply:
I see, sorry I did not notice, thank you!